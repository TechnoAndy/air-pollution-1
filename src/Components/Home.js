import { useEffect } from 'react';
import { Button, Container } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import { NavLink } from 'react-router-dom';
import { fetchCountries, filteredCountries } from '../Redux/Reducers/Countries';

const Home = () => {
  let countryName;

  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countryReducer);

  useEffect(() => {
    dispatch(fetchCountries());
  }, [dispatch]);

  return (
    <div className="home">
      <div className="searchBoxOuter">
        <input
          className="countrySearch"
          type="text"
          placeholder="Country Name"
          value={countryName}
          onChange={(e) => {
            countryName = e.target.value;
          }}
        />
        <Button
          type="button"
          onClick={async () => {
            await dispatch(fetchCountries());
            dispatch(filteredCountries(countryName));
          }}
        >
          Search
        </Button>
      </div>
      <div className="home-heading">
        <header>
          <div className="map-image">
            <img src="./africa.png" alt="Map Of Africa" />
          </div>
          <div className="map-heading">
            <h1>
              Quality of air in
              <br />
              Africa.
            </h1>
          </div>
        </header>
      </div>

      <Container className="home-countries">
        {countries.map((country) => (
          <div key={uuidv4()} className="countryInfo">
            <div className="card-t">
              <div className="home-card-heading">{country.name}</div>
              <div className="card-body">
                {country.countrycode}
                <br />
                {country.region}
              </div>
              <button type="submit">
                <NavLink
                  className="nav-link "
                  to="/city"
                  state={{
                    info: country,
                  }}
                >
                  Visit Cities
                </NavLink>
              </button>
            </div>
          </div>
        ))}
      </Container>
    </div>
  );
};

export default Home;
