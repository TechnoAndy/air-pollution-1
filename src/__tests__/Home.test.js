import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/ConfigureStore';
import Home from '../Components/Home';

it('Home', () => {
  const utils = render(
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path="/" element={<Home />} />
        </Routes>
      </Router>
    </Provider>,
  );
  expect(utils).toMatchSnapshot();
});
