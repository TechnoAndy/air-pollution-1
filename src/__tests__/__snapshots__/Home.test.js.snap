// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <div
          class="countrySearchBox"
        >
          <input
            class="countrySearch"
            placeholder="Country Name"
            type="text"
            value=""
          />
          <button
            class="btn btn-secondary"
            type="button"
          >
            Search
          </button>
        </div>
        <div
          class="home-heading"
        >
          <header>
            <div
              class="map-image"
            >
              <img
                alt="Map Of Africa"
                src="./africa.png"
              />
            </div>
            <h1>
              Quality of air in
              <br />
              Africa.
            </h1>
          </header>
        </div>
        <div
          class="home-countries container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <div
        class="countrySearchBox"
      >
        <input
          class="countrySearch"
          placeholder="Country Name"
          type="text"
          value=""
        />
        <button
          class="btn btn-secondary"
          type="button"
        >
          Search
        </button>
      </div>
      <div
        class="home-heading"
      >
        <header>
          <div
            class="map-image"
          >
            <img
              alt="Map Of Africa"
              src="./africa.png"
            />
          </div>
          <h1>
            Quality of air in
            <br />
            Africa.
          </h1>
        </header>
      </div>
      <div
        class="home-countries container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
